### ISSUE TITLE
Email OTP should have a period of validity

### ISSUE THREAD

**Issue by bil0u** [mkalioby/django-mfa2#66]
Hi,

I searched for this functionnality in the readme and by exploring the code and it seems that there is no concept of period of validity involved in the logic of [`mfa.Email.auth`](https://github.com/mkalioby/django-mfa2/blob/4903967c23e7dc515090c26bcacec2b64d8dbe9d/mfa/Email.py)

Maybe this could be implemented using the request session only for email keys, or a more generic approach could be added directly using [`User_Keys.expires`](https://github.com/mkalioby/django-mfa2/blob/7a154cfa340819363f1f076e7d96b4464a56ae8a/mfa/models.py#L16) and a middleware.

In both case, this timeout should be configurable per OTP method and have sensible defaults.

Happy to discuss it further if you think it's worth it !

**Comment by mkalioby** [comment]
Good point, but it is only the email method which needs this implementation.

Do you like to give it a try or shall i do it?

**Comment by bil0u** [comment]
Either way is fine, I can have a look during the week !

**Comment by mkalioby** [comment]
Sure.  Let me know if you need help.
